### PR TITLE
Systems: documentation and minor changes

### PR DESCRIPTION
**Pull request by schillic**
I changed a bit the documentation of `Systems.jl`.

The only code change are
- I turned some inner constructors to outer constructors.
- I reordered the fields of `DiscreteSystem`.
- I changed the field name of `InputState` from `sf` to `set`. (This implied a lot of changes in other files.)

### CODE REVIEW

**Comment by schillic** [comment]
I noticed that we might want to change the way the `InputState` iterator works.
We create a new instance for each time step (for `TimeVaryingNonDeterministicInput` only, but still).

What about instead changing this wrapper such that
* it keeps the whole array
* updates its index internally (i.e., it is a mutable type)
* always returns the respective set corresponding to its current index

Then instead of calling `input_state.sf`, we call a function, say, `get(input_state)`.
For the array-backed iterator we could even consider just returning a view.

File: src/Systems/Systems.jl [hunk 1]
```diff
@@ -15,107 +15,156 @@ export AbstractSystem,
...
 where:
 
-- A is a square matrix
-- x(0) ∈ X0
-- u(t) ∈ U(t), where U(t) is a piecewise-constant set-valued function defined
-  over [t1, t1+δ], ... , [tN, tN+δ]
+- ``A`` is a square matrix
+- ``x(0) ∈ X0``
+- ``u(t) ∈ U(t)``, where ``U(t)`` is a piecewise-constant set-valued function
+  defined over ``[t1, t1+δ], ... , [tN, tN+δ]`` for some δ
```
Comment (mforets) [inline src/Systems/Systems.jl]: the fact that there is no `delta`as a field of this type, but it's still mentioned in the docstring, is a bit misleading, i acknowledge.. maybe we should just say (quoting decoPaper):

> where ``U(\cdot)`` is a piecewise-constant set-valued function, i.e. we consider a possibly time-varying discrete sequence ``\{U(k)\}_k``

File: src/Systems/Systems.jl [hunk 2]
```diff
@@ -132,34 +181,58 @@ end
...
 
 where
 
-- A is a square matrix
-- x(0) ∈ X0
-- u_{k} ∈ U_{k}, where U_{k} is a piecewise-constant set-valued function defined over [t1, t1+δ], ... , [tN, tN+δ]
+- ``A ``is a square matrix
+- ``x(0) ∈ X0``
+- ``u_{k} ∈ U_{k}``, where ``U_{k}`` is a piecewise-constant set-valued function
+  defined over ``[t1, t1+δ], ... , [tN, tN+δ]`` for some δ
```
Comment (mforets) [inline src/Systems/Systems.jl]: again we don't support/handle the time shifts `t_1` at all (yet), so this is more properly ``[0, \delta], ..., [(N-1)\delta, N\delta]``.

File: src/Systems/Systems.jl [hunk 3]
```diff
@@ -15,107 +15,156 @@ export AbstractSystem,
...
 
 """
-Type that represents a constant non-deterministic input.
+Type that represents a constant nondeterministic input.
+
+The iteration over this set is such that its `state` is a tuple
+(`set`, `index`), where `set` is the value of the input, represented as a
+`LazySet`, and `index` counts the number of times this iterator was called. Its
+length is infinite, since the input is defined for all times. The index of the
+input state is always contantly 1.
```
Comment (mforets) [inline src/Systems/Systems.jl]: constantly

File: src/Systems/Systems.jl [hunk 4]
```diff
@@ -15,107 +15,156 @@ export AbstractSystem,
...
 Abstract type representing a nondeterministic input. The input can be either constant
 or time-varying. In both cases it is represented by an iterator.
 """
 abstract type NonDeterministicInput end
 
+
 """
-Type that represent the state of a constant or time-varying input, as a pair of
-sf (set represented by its support function), and an index.
+Type that represent the state of a `NonDeterministicInput`.
```
Comment (mforets) [inline src/Systems/Systems.jl]: represents

**Comment by mforets** [comment]
> I noticed that we might want to change the way the InputState iterator works.
the ideas sound good!  do you create a new issue?

**Comment by schillic** [comment]
Thanks for the review, and I added issue #21.

**Pull request merged**